### PR TITLE
[SHELL32][SDK] Implement SHIsBadInterfacePtr

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Implementing missing features...
JIRA issue: [CORE-9277](https://jira.reactos.org/browse/CORE-9277)

## Proposed changes

- Implement `SHIsBadInterfacePtr` function in `dll/win32/shell32/misc.cpp`.
- Add `SHIsBadInterfacePtr` prototype into `<undocshell.h>`.

### CODE REVIEW

File: dll/win32/shell32/wine/shellole.c [hunk 1]
```diff
@@ -810,3 +810,13 @@ HRESULT WINAPI SHCreateQueryCancelAutoPlayMoniker(IMoniker **moniker)
     if (!moniker) return E_INVALIDARG;
     return CreateClassMoniker(&CLSID_QueryCancelAutoPlay, moniker);
 }
+
+/*************************************************************************
+ *  SHIsBadInterfacePtr [SHELL32.84]
+ */
+BOOL WINAPI SHIsBadInterfacePtr(IUnknown *punk, UINT cbVtbl)
```
Comment (HBelusca) [inline dll/win32/shell32/wine/shellole.c:817]: You've put it in a wine-synced file, but this function is not implemented in wine. Please either move it somewhere else, or surround its implementation in `#ifdef __REACTOS__` .

Comment (katahiromz) [reply dll/win32/shell32/wine/shellole.c:818]: Added `#ifdef __REACTOS__` ... `#endif`.

**Comment by learn-more** [comment]
Do you need this function for anything?
Or did you find some random application depending on it?

**Comment by katahiromz** [comment]
> Do you need this function for anything?
> Or did you find some random application depending on it?

Nope. Nope.

File: dll/win32/shell32/wine/shellole.c (hunk 1 above)
Comment (tkreuzer) [reply dll/win32/shell32/wine/shellole.c:818]: Please don't put new functions into wine synced files.

Comment (katahiromz) [reply dll/win32/shell32/wine/shellole.c]: Moved to `misc.cpp`.

**Comment by JoachimHenze** [comment]
`Nope. Nope.`
We had a rule once to only implement what we really needed for something or where we could prove that any 3rd party app depends on. I always found that to be a very good rule. I therefore recommend to not merge this PR 'just because we can'. But to wait instead until such a need would evolve. It has a proper description so we would be able to find it later.

**Comment by katahiromz** [comment]
I found it was needless. Closed.

**Comment by learn-more** [comment]
> `Nope. Nope.` We had a rule once to only implement what we really needed for something or where we could prove that any 3rd party app depends on. I always found that to be a very good rule. I therefore recommend to not merge this PR 'just because we can'. But to wait instead until such a need would evolve. It has a proper description so we would be able to find it later.

Really now, where was that rule defined?

**Pull request closed without merging**